### ISSUE TITLE
webpack-dev-server misses most Vim write events

### ISSUE THREAD

**Issue by michaeldye** [webpack/webpack#781]
When writing (:w) files in Vim on Linux, ```chokidar```'s example script produces events like this:

```
RawEvent: change Quota.js { watchedPath: '/project/app/modules' }
change /project/app/modules/Quota.js
```
... inotify in Linux reports the event this way:
```
CLOSE_NOWRITE,CLOSE on /project/app/modules/Quota.js at 1423591700
```

Unfortunately, almost every file update is missed by an idle webpack-dev-server instance (```webpack-dev-server --watch --hot --inline```). It seems the first write is reliably detected and only sporadic others are. 

I'm using chokidar@1.0.0-rc3 <- watchpack@0.2.3 <- webpack@1.5.3

See also paulmillr/chokidar#226

**Comment by acthp** [comment]
Also have this issue. It's unusable.

**Comment by bloodyowl** [comment]
:+1:

**Comment by pikeas** [comment]
Is this related to https://github.com/webpack/webpack/issues/793#issuecomment-75069334?

**Comment by michaeldye** [comment]
@pikeas Doesn't seem related to me, we're using different OSes with different filesystems.

**Comment by pikeas** [comment]
@michaeldye Gotcha. I thought it might have been a Linux system running in a Vagrant VM (eg, code dir loaded in as a volume).

**Comment by virajsanghvi** [comment]
Just wanted to mention that this happens with --watch as well as the dev server.

Does anyone have any workarounds?

**Comment by virajsanghvi** [comment]
Thought I would mention that using OldWatchingPlugin actually solved my issues with watching vim file changes (not using the dev-server).

**Comment by acthp** [comment]
Can you post more details? I need a workaround.

**Comment by virajsanghvi** [comment]
I just added the plugin to my webpack.config.js:

```
plugins: [
  ...
  new webpack.OldWatchingPlugin(),
  ...
],
```

According to #125, OldWatchingPlugin should theoretically be giving me issues, but, at least for webpack 1.6, I haven't had any issues with it in comparison to NewWatchingPlugin (which is now the default), which effectively broke watching (for me). Anyways, YMMV.

**Comment by wingedkiwi** [comment]
This really makes it completely unusable for me :-/

**Comment by wingedkiwi** [comment]
Found the problem: 
You have to tell vim how it should save the file. On most systems it is set to "auto" letting vim decide the saving strategy. Sometimes it writes directly to the file, sometimes it renames the old file and write a new one. The latter one causes watching problems.

Simple set ":set backupcopy=yes" in vim and it works perfectly!

See: http://vimdoc.sourceforge.net/htmldoc/options.html#'backupcopy'

**Comment by acthp** [comment]
@wingedkiwi Thank you so much! Working perfectly now.

This really should be a vim faq thing: using vim with a file watcher. Seems like this issue could be closed, or maybe a brief note added to the watcher docs, to warn other vim users.

**Comment by jhnns** [comment]
I'm glad you have found a fix :)

**Comment by jhnns** [comment]
The docs are editable. So if you want to give your VIM colleagues a hint, go for it :+1:

**Comment by gaogao1030** [comment]
@wingedkiwi Thank you  ありかとう　ございました　非常感谢

**Comment by pizza2code** [comment]
@jhnns Should this have been closed?  It's unusual that changing our source editor's file-saving method would be considered a fix, rather than a **workaround** for an existing issue.  All other file watchers I've used (gulp.watch() and watchify) function fine without changing the behavior of my file editors.

**Comment by jhnns** [comment]
I can re-open it, but we need a champion for this from the VI community, otherwise nothing will happen. Every editor I know works like expected. Do you want to contribute @pizza2code?

**Comment by aethant** [comment]
I've tried applying the workaround mentioned here in vim `:set backupcopy=yes` (vim 7.4.1755) on Mac OS X 10.10.5 (Yosemite) and I get *maybe* 3-4 (usually 1) saves before Webpack stops responding. Atom _does_ save and get Webpack to respond, but not after I've opened the same file in vim. I have to restart Webpack to get it working. I'm also using Webpack inside a Docker container in this instance. 

Anyone have any ideas why in this case the workaround does work?
